pos_restricted_customer_list/static/src/js/pos_restricted_customer_list.js

```diff
@@ -1,25 +1,25 @@
-'use strict';
+odoo.define('pos_restricted_customer_list.point_of_sale.models', function (require) {
+    'use strict';
 
-openerp.pos_restricted_customer_list = function (instance) {
-	var module = instance.point_of_sale;
+    var PosModels = require('point_of_sale.models');
+    var PosModel = PosModels.PosModel;
+    var PosModelSuper = PosModel.prototype;
 
-	// we can't extend it because self.pos not ready yet
-	var _initializePosModel_ = module.PosModel.prototype.initialize;
-	module.PosModel.prototype.initialize = function(session, attributes){
-		// override domain for res.partner to limit customers loaded
-		this.models.some(function (m, idx) {
-			if (m.model !== 'res.partner')
-				return false;
-			// check if not already done by someone else
-            for(var i = 0; i < m.domain.length; i++) {
-                var domain_tuple = m.domain[i];
-                if (domain_tuple[0] === 'available_in_pos') {
-                    return true; // domain already added
+    PosModels.PosModel = PosModel.extend({
+        initialize: function (session, attributes) {
+            var self = this;
```

Comment:
I'd say this isn't needed

Reply:
I prefer to always write `var self = this;` in case it's useful in the future. It doesn't cost anything.

Reply:
Well, you don't put code for the just in case :sweat_smile: If it's useless now, remove it, as it makes it easier to read.